### PR TITLE
Wrote GenServer blog post

### PR DESCRIPTION
**Pull request by kelvinma**

### CODE REVIEW

File: _posts/2018-03-16-first-steps-into-gen-servers.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title:  "Automating Processes with Gen Servers: A Beginner's Approach"
+date:   2018-03-16 10:00:00 -0400
+categories: elixir automation microservices
+author: Kelvin Ma
+---
+As a financial services company, Forward Financing invests a signficant amount of time and resources around timebound reporting -- think daily, weekly, monthly, etc., report generation. As opportunities for automation are concerned, it doesn't get more obvious than this.
+
+A big part of my growth as an engineer comes from getting out of my comfort zone, using new and unfamiliar tools to build hopefully useful tools. After doing a little research on potential solutions for our automation opportunity, I settled on Elixir's GenServer, or Generic Server, pattern. Coming from a primarily Object Oriented Programming background with Rails, this pattern was a bit foreign to me, to say the least.
```
Comment (bradleyden) [inline _posts/2018-03-16-first-steps-into-gen-servers.md]: The first sentence of this paragraph is a bit awkward. I'd drop 'hopefully' and only use 'tools' once. Maybe 'using new and unfamiliar technologies to build useful tools'?

File: _posts/2018-03-16-first-steps-into-gen-servers.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+---
+layout: post
+title:  "Automating Processes with Gen Servers: A Beginner's Approach"
+date:   2018-03-16 10:00:00 -0400
+categories: elixir automation microservices
+author: Kelvin Ma
+---
+As a financial services company, Forward Financing invests a signficant amount of time and resources around timebound reporting -- think daily, weekly, monthly, etc., report generation. As opportunities for automation are concerned, it doesn't get more obvious than this.
```
Comment (bradleyden) [inline _posts/2018-03-16-first-steps-into-gen-servers.md]: I think 'invests ... time and resources IN timebound reporting' sounds a little cleaner to me.

File: _posts/2018-03-16-first-steps-into-gen-servers.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+No controllers. No models. No database.
+
+Doesn't get much simpler than that!
+
+### Future Implications
+
+At this point, you know that I can now read and follow basic documentation. What a relief!
+
+What I find most fascinating about this particular pattern is its ability to effectively manage multiple microservices within a single supervision tree. While in this particular example I'm only implementing a single GenServer, adding another two, three...10, more, would be a trivial process. And they could even communicate with each other with their own messages.
```
Comment (bradleyden) [inline _posts/2018-03-16-first-steps-into-gen-servers.md]: I think you can drop the commas before and after 'more'? I didn't go to j-school so correct me if I'm wrong ;)

also, 'communicate with each other with their own messages' is a little awkward with the double 'with'. Maybe 'using their own messages'?

File: _posts/2018-03-16-first-steps-into-gen-servers.md (hunk 2 above)
Comment (kelvinma) [reply _posts/2018-03-16-first-steps-into-gen-servers.md]: 👍

File: _posts/2018-03-16-first-steps-into-gen-servers.md (hunk 3 above)
Comment (kelvinma) [reply _posts/2018-03-16-first-steps-into-gen-servers.md]: 👍

File: _posts/2018-03-16-first-steps-into-gen-servers.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+From the [GenServer](https://hexdocs.pm/elixir/GenServer.html) HexDocs:
+
+> A GenServer is a process like any other Elixir process and it can be used to keep state, execute code asynchronously and so on.
+
+Seems pretty straightforward. By writing a GenServer and isolating this single task into its own process, I am effectively creating a microservice that can be managed by the application's own [supervision tree](https://hexdocs.pm/elixir/Supervisor.html).
+
+Here's what I ended up with, using the default callbacks for GenServer:
+
+```
```
Comment (abhishekjain16) [inline _posts/2018-03-16-first-steps-into-gen-servers.md]: @kelvinma Can we use syntax highlighting like - https://jekyllrb.com/docs/posts/#highlighting-code-snippets
We might have elixir syntax highlighting and it is inbuilt in jeykll.

Comment (kelvinma) [reply _posts/2018-03-16-first-steps-into-gen-servers.md]: WOOT! Pretty!

**Pull request merged**